### DISCUSSION TITLE
Warning: trim() expects parameter 1 to be string, array given

### DISCUSSION THREAD

**Discussion by alaindev** [PHPOffice/PhpSpreadsheet#2778] [General]
when cell with indirect formula "=SUM(D1:INDIRECT(ADDRESS(ROW();COLUMN()-4)))"

 Warning: trim() expects parameter 1 to be string, array given in C:\Program Files (x86)\EasyPHP-Devserver-17\eds-www\aplitec\includes\modules\phpoffice\phpspreadsheet\src\PhpSpreadsheet\Calculation\Calculation.php on 

Patch : Calculation.php line 4536 & 4546
if ((trim($operand1Data['value']) != '') && (is_numeric($operand1Data['value']))) {
>>> should be 
if (!is_array($operand1Data['value']) && (trim($operand1Data['value']) != '') && (is_numeric($operand1Data['value']))) {

[relativecell.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/8554565/relativecell.xlsx)

Regards
Alain

**Comment by alaindev** [comment]
Sorry
I forgot PHP Code & version : PHPSpreadsheet 1.23

Appears at save
<?php
	require_once (DIR_WS_MODULES . 'autoload.php'); // ade MODE TO php7.4 AND PHPspreadsheet
	
	$file = 'c:\tmp\relativecell.xlsx';
	$showPHPExcel = \PhpOffice\PhpSpreadsheet\IOFactory::load($file);
	$sheets_array = $showPHPExcel->getSheetNames();
	$liassewriter = \PhpOffice\PhpSpreadsheet\IOFactory::createWriter($showPHPExcel, "Xlsx");
	$file = 'c:\tmp\file1.xlsx';
	$liassewriter->save($file);
?>

**Comment by oleibman** [comment]
PR #2774 makes minor changes to the same area of code. I will try to address this in that change. In the meantime, can you open this as a new issue rather than a discussion topic.

**Comment by alaindev** [comment]
Thanks
    created the issue 2780
    Alain
    
    Le 25/04/2022 à 16:50, oleibman a
      écrit :
    
    
      
      PR #2774 makes minor changes to the
        same area of code. I will try to address this in that change. In
        the meantime, can you open this as a new issue rather than a
        discussion topic.
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you authored the thread.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/PHPOffice/PhpSpreadsheet/discussions/2778#discussioncomment-2630603",
"url": "https://github.com/PHPOffice/PhpSpreadsheet/discussions/2778#discussioncomment-2630603",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by oleibman** [comment]
Fixed by PR #2774.
